### ISSUE TITLE
Allow mixing of pre-processing and feature extraction modules?

### ISSUE THREAD

**Issue by damellis** [nickgillian/grt#61]
There are times when it might be useful to mix these, e.g. taking the average over time of the result of an FFTFeatures module. Is there a reason that pre-processing and feature extraction need to be distinct? Aren't they both just functions from a vector<double> to another vector<double>?

**Comment by nickgillian** [comment]
Yes, I've been meaning to improve this as there are many cases when you want to mix both.  The main logic for the original setup is that a pre-processing module is something that keeps the dimensionality the same (so the number of inputs/outputs will always match), whereas a feature extraction module could have varying dimensionality (so a 3 dimensional input vector could result in a 10 dimensional feature vector for instance)....but I agree it makes more sense to have a more generic ProcessingNode (or something) which can be chained together in the pipeline and consist of either pre-processing or feature extraction algorithms.

I'll look at reconfiguring this over the weekend.
